sources/packages/web/src/components/common/students/StudentFileUploads.vue

```diff
@@ -0,0 +1,206 @@
+                  ></span>
+                  <span>{{ slotProps.data.fileName }}</span>
+                </div>
+                <div v-else>
+                  <span class="mr-4">
+                    <v-icon icon="fa:far fa-file-alt" size="20"></v-icon
+                  ></span>
+                  <span>{{ slotProps.data.fileName }}</span>
+                </div>
+              </template></Column
+            >
+          </DataTable>
+        </toggle-content>
+      </content-group>
+      <formio-modal-dialog
+        :max-width="730"
+        ref="fileUploadModal"
+        title="Upload file"
+        :formData="initialData"
+        formName="uploadStudentDocumentsAEST"
+      >
+        <template #actions="{ cancel, submit }">
+          <v-row class="m-0 p-0">
+            <v-btn color="primary" variant="outlined" @click="cancel"
+              >Cancel</v-btn
+            >
+            <check-permission-role :role="Role.StudentUploadFile">
+              <template #="{ notAllowed }">
+                <v-btn
+                  class="float-right"
+                  @click="submit"
+                  color="primary"
+                  variant="elevated"
+                  :disabled="notAllowed"
+                  >Upload now</v-btn
+                >
+              </template>
+            </check-permission-role>
+          </v-row>
+        </template>
+      </formio-modal-dialog>
+    </body-header-container>
+  </tab-container>
+</template>
+
+<script lang="ts">
+import { onMounted, ref, defineComponent } from "vue";
+import {
+  DEFAULT_PAGE_LIMIT,
+  FormIOForm,
+  PAGINATION_LIST,
+  LayoutTemplates,
+  Role,
+} from "@/types";
+import { StudentService } from "@/services/StudentService";
+import {
+  useFormatters,
+  useFileUtils,
+  ModalDialog,
+  useFormioUtils,
+  useSnackBar,
+} from "@/composables";
+import FormioModalDialog from "@/components/generic/FormioModalDialog.vue";
+import { AESTFileUploadToStudentAPIInDTO } from "@/services/http/dto/Student.dto";
+import CheckPermissionRole from "@/components/generic/CheckPermissionRole.vue";
+
+export default defineComponent({
+  components: {
+    FormioModalDialog,
+    CheckPermissionRole,
+  },
+  emits: ["loaded"],
+  props: {
+    studentId: {
+      type: Number,
+      required: true,
+    },
+    canUploadFiles: {
+      type: Boolean,
+      required: true,
+    },
+    canDownloadFiles: {
+      type: Boolean,
+      required: true,
+    },
+    studentFileUploads: {
+      type: Array,
+      required: true,
+    },
+  },
+  setup(props: any, context) {
+    const fileUploadModal = ref({} as ModalDialog<FormIOForm | boolean>);
+    const { dateOnlyLongString } = useFormatters();
+    const snackBar = useSnackBar();
+    const fileUtils = useFileUtils();
+    const formioUtils = useFormioUtils();
+    const initialData = ref({ studentId: props.studentId });
+
+    const loadStudentFileUploads = async () => {
```

Comment:
I would say that the calling service(API) part can be inside the component itself. Because getStudentFiles can be one service which can handle the API call for all the client types. And as per our latest dev discussion we can have a common model returned from service for the Vue perspective.

```ts
const loadStudentFileUploads = async () => {
      studentFileUploads.value =
        await StudentService.shared.getStudentFiles(props.studentId);
    };
```

StudentService.ts

```ts
async getStudentFiles(
    studentId: number,
  ): Promise<StudentFile[]> {
   // Transformation if needed
    return ApiClient.Students.getStudentFiles(studentId);
  }
```

Reply:
@dheepak-aot `getStudentFiles` method in StudentService.ts is the service for student-side file uploader logic. 
![Uploading image.png…]()
 This common component is for Institution and AEST datatable

Reply:
But as @dheepak-aot  mentioned we can call the service from here 
![image](https://user-images.githubusercontent.com/77353155/233169159-4655249e-d5d7-490b-923e-c83888a37c91.png)
you can refer to other shared components. and we already have the service probably you can rename and adjust it 
![image](https://user-images.githubusercontent.com/77353155/233169484-458cc8a2-57f3-48fe-8e12-14da4327ff5a.png)

Reply:
I have made the suggested change.